### PR TITLE
[CALCITE-6251] innerEnumerator in EnumerableDefaults::correlateBatchJoin is not closed

### PR DESCRIPTION
**Pull request by kramerul**

### CODE REVIEW

**Comment by rubenada** [comment]
@kramerul  CI is failing due to commit message:
```
<[invalid git log message '[CALCITE-6251] innerEnumerator in EnumerableDefaults::correlateBatchJoin is not closed'; Message must start with upper-case letter]>
```

I think the simplest would be to amend the commit message and start with uppercase: `[CALCITE-6251] InnerEnumerator in ...`

**Comment by kramerul** [comment]
I had to fix also another issue with `@Nullable`

**Comment by rubenada** [comment]
LGTM

File: linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java [hunk 1]
```diff
@@ -1799,11 +1801,16 @@ private boolean innerHasNext() {
             i = -1;
           }
 
-          @Override public void close() {
-            outerEnumerator.close();
+          private void closeInner() {
```
Comment (JiajunBernoulli) [inline linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1804]: Should we use `synchronized`?

Comment (rubenada) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1804]: Interesting point. 
Maybe I'm wrong, but my understanding is that our enumerators are not thread-safe, so if a 3rd party uses them on a multi-threaded environment, they would need to handle synchronization on their end.

Comment (JiajunBernoulli) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:1804]: Get it

**Review by JiajunBernoulli — APPROVED** [review]

**Pull request merged**